raster/src/main/scala/geotrellis/raster/interpolation/KrigingInterpolation.scala

```diff
@@ -0,0 +1,62 @@
+package geotrellis.raster.interpolation
+
+import geotrellis.raster._
+import geotrellis.vector._
+import spire.syntax.cfor._
+
+abstract sealed class KrigingInterpolationMethod
+case object KrigingSimple extends KrigingInterpolationMethod
+case object KrigingOrdinary extends KrigingInterpolationMethod
+case object KrigingUniversal extends KrigingInterpolationMethod
+case object KrigingGeo extends KrigingInterpolationMethod
+
+
+object KrigingInterpolationMethod {
+  val DEFAULT = KrigingOrdinary
+}
+
+abstract class KrigingInterpolation(method: KrigingInterpolationMethod, points: Seq[PointFeature[Int]], re: RasterExtent,
```

Comment:
@lossyrob Thanks for the feedback, I will change my code to reflect a more functional oriented approach. I am about to finish one of the Kriging modules(Simple); and then I will switch to spire to do the matrix computations.